### PR TITLE
MAT-7192 added action center and delete button to library details screen

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-7192](https://jira.cms.gov/browse/MAT-7192 )
(Optional) Related Tickets:

### Summary
added action center and delete button to library details screen

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/common/useFormikResetOnEvent.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+
+// reusable component to hook into form locations.
+// We have a discard dialog in the library-actions-center that needs to know about every other form in the app, and trigger the reset onContinue
+// We could have that onContinue logic emit an event, and this component should add the listener on each form that needs to blank.
+const useFormikResetOnEvent = (formik) => {
```
Comment (sb-cecilialiu) [inline src/components/common/useFormikResetOnEvent.ts:6]: Would be nice to make it globally reusable, instead of having a separate one in madie-cql-library and madie-measure. Maybe in the future.

File: src/components/editCqlLibrary/EditCqlLibrary.tsx [hunk 2]
```diff
@@ -143,6 +160,28 @@ const EditCqlLibrary = () => {
     });
   };
 
+  const deleteDraftLibrary = async (id: string) => {
+    cqlLibraryServiceApi
+      .deleteDraft(id)
```
Comment (sb-cecilialiu) [inline src/components/editCqlLibrary/EditCqlLibrary.tsx:165]: Can you please add some test coverage for delete draft? Thanks!

File: src/components/common/useFormikResetOnEvent.ts (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/components/common/useFormikResetOnEvent.ts:6]: Well we already have this for madie-measure in its component so I just created a separate component for this in cql library and I don't think we will use it any further in any other place so it should be okay to leave it here

File: src/components/editCqlLibrary/EditCqlLibrary.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/editCqlLibrary/EditCqlLibrary.tsx:165]: I think I forgot to add that file to the PR. I just added it thanks for pointing that

Comment (sb-cecilialiu) [reply src/components/editCqlLibrary/EditCqlLibrary.tsx:165]: @sb-prateekkeerthi great thanks!

**Review by sb-cecilialiu — APPROVED** [review]
